moveit_py/src/moveit/moveit_core/kinematic_constraints/utils.cpp

```diff
@@ -76,6 +76,7 @@ moveit_msgs::msg::Constraints construct_link_constraint(const std::string& link_
     quaternion.quaternion.x = orientation.value()[0];
     quaternion.quaternion.y = orientation.value()[1];
     quaternion.quaternion.z = orientation.value()[2];
+    quaternion.quaternion.w = orientation.value()[3];
```

Comment:
Because this has got me in the past, I would be remiss not to ask: Please double-check that the orientation vector coming in here is actually XYZW and now WXYZ. Different tools / places use different conventions.

This is where unit tests are handy too.

Reply:
I can add the specification of this to the API documentation. It is currently a user provided list/array. I was assuming the convention XYZW as I believe this aligns with the convention in http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Quaternion.html. 

To make this more fault tolerant we could enforce that the user passes a Python message type (essentially a python object). This makes the Python code a bit more concise but adds bloat. 

What are your thoughts?

Reply:
I think documenting the convention is fine. I personally would like to *not* use ROS messages where possible in the hope that maybe someday MoveIt will be usable standalone